test/type/number_test.exs

```diff
@@ -0,0 +1,140 @@
+defmodule Litmus.Type.NumberTest do
+  use ExUnit.Case
+
+  alias Litmus.Type
+
+  describe "validate_field/3" do
+    test "validates property values of data based on their Number schema definition in Type.Number module" do
+      field = "id"
+      data = %{"id" => 1}
+
+      type = %Type.Number{
+        required: true
+      }
+
+      assert Type.Number.validate_field(type, field, data) == {:ok, data}
+    end
+  end
+
+  describe "convert string to number" do
+    test "returns :ok with modified value when convert is true" do
+      data = %{"id" => ".6"}
+      data_1 = %{"id" => "6"}
```

Comment:
let's use more descriptive names than `data` and `data_1` when we have multiple maps.